docs/Developer Documentation/ResumeController/index.md

```diff
@@ -0,0 +1,205 @@
+In case if any  other application already registered, StateController will take care about hmi_states conflicts resolving. 
+
+
+### Data resumption
+
+SDL restore application data if application send appropriate `hash` in RAI. Hash updates after each data change. 
+SDL stores resumption data either in json or in database, this option is configurable. 
+
+ResumeControllerImpl request app data from `ResumptionData` class and provide it to `ResumptionDataProcessor`
+
+`ResumptionDataProcessor` is responcible for restoring application data and provide result to RAI via callback
+
+|||
+Figure 2: Resumption data seauence Overview
+![Resumption data seauence](./assets/data_resumption.png)
+|||
+
+
+### ResumptionData
+
+`ResumptionData` class user to represend resumption data agnostic to data storage. 
+`ResumptionData` provides app resumption data as smart objects. 
+
+|||
+Figure 2: Resumption data classes
+![Resumption data classes](./assets/resumption_data_class.png)
+|||
+
+There are 2 implementations of resumption data : 
+ * `ResumptionDataJson`
+ * `ResumptionDataDB`
+
+ `ResumptionData` does not contains and active components : timers, reactions, callbacks, etc ...
+ It behaves like data a storage.
+
+### ResumptionDataProcessor
+
+ResumptionDataProcessor responcible for restoring resumption data and track it's status. 
+
+Main public function for resumptions is `ResumptionDataProcessor::Restore` : 
+```cpp
+  /**
+  * @brief Running resumption data process from saved_app to application.
+  * @param application application which will be resumed
+  * @param saved_app application specific section from backup file
+  */
+  void Restore(app_mngr::ApplicationSharedPtr application,
+               smart_objects::SmartObject& saved_app,
+               ResumeCtrl::ResumptionCallBack callback);
+```
+
+`ResumeCtrl::ResumptionCallBack callback` is a function that should be called after data resumption : 
+```cpp
+typedef std::function<void(mobile_apis::Result::eType result_code,
+                             const std::string& info)> ResumptionCallBack;
+```
+
+Some resumption data should be restored in `Application` class it self.
+Some resumption data should stored in pligins : ApplicationExtensions. 
+Some resumption data requires sending HMI request. 
+
+ResumptionDataProcessor is inherited from EventObserver to track responses. 
+
+If responses are succesfull ResumptionDataProcessor just call `callback(SUCCESS)`
+
+If part of the data was failed to restore, ResumptionDataProcessor should revert already restored data and call  `callback(ERROR_CODE, info)`.
+
+Requirenments available in proposal [Handle response from HMI during resumption data](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0190-resumption-data-error-handling.md)
+
+RAI will wait for callback to send response to mobile App.
```

Comment:
```suggestion
RAI will wait for callback to send response to a mobile application.
```

Reply:
I believe we should keep the document in a consistent state: above mentioned just `application`, but here we have `mobile application`. We need to choose one of variants: `application` or `mobile application` and use it in each place that we need.